### ISSUE TITLE
ChannelType mismatch

### ISSUE THREAD

**Issue by sthaiya** [heroiclabs/nakama-java#33]
While trying to join a group through the nakama-java client on android i received the Error **User ID not found**. On investigating i realised a mismatch in the enum classes. 

rtapi,ChannelType on the server side define:

	ChannelJoin_TYPE_UNSPECIFIED ChannelJoin_Type = 0
	ChannelJoin_ROOM ChannelJoin_Type = 1
	ChannelJoin_DIRECT_MESSAGE ChannelJoin_Type = 2
	ChannelJoin_GROUP ChannelJoin_Type = 3

while the java client defines 

   ROOM(0),
    DIRECT_MESSAGE(1),
    GROUP(2);

Thus when you try to join a GROUP from the client side the serve receives const 2 which is a DIRECT_MESSAGE

**Comment by mofirouz** [comment]
Very good catch. Could you kindly open a PR for this?

**Comment by mofirouz** [comment]
This probably is the case on other Enum types too.

**Comment by sthaiya** [comment]
> Very good catch. Could you kindly open a PR for this?

WIll do so though I am unable to compile the code due to dependency problems as I raised in a previous issue. Will try looking through all other enums too

**Comment by lugehorsam** [comment]
This is addressed by #35 which defines the following:

```message ChannelJoin {
  // The type of chat channel.
  enum Type {
    // Default case. Assumed as ROOM type.
    TYPE_UNSPECIFIED = 0;
    // A room which anyone can join to chat.
    ROOM = 1;
    // A private channel for 1-on-1 chat.
    DIRECT_MESSAGE = 2;
    // A channel for group chat.
    GROUP = 3;
  }```

**Comment by karan1213** [comment]
In the latest codebase, I still see this mismatch. Is this issue really fixed in #35
